packages/flutter/lib/src/material/text_field.dart

```diff
@@ -371,6 +381,38 @@ class TextField extends StatefulWidget {
   /// textfield's internal gesture detector, use a [Listener].
   final GestureTapCallback onTap;
 
+  /// Function to generate a custom counter widget. Passed the build context,
+  /// current and max lengths of the input, and [isFocused]. Will be placed
+  /// below the line in place of the default widget built when passing
+  /// [counterText].
+  ///
+  /// Accessibility will be up to you if you use this instead of [counterText].
+  /// Be sure to wrap your widget in a [Semantics] widget and use the given
```

Comment:
... Maybe we should just add the Semantics widget to the value returned by buildCounter?

Reply:
That's an interesting idea haha.  Could there be a situation when the user wouldn't want us to do that?  If not then I should add it.